src/varint.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "common.h"
+#include "varint.h"
+
+uintmax_t decode_varint(const unsigned char **bufp)
```

Comment:
Let's make sure we have a `git_` prefix on all the functions that are scoped larger than a compilation unit.  Even when this is not explicitly exported, it could end up that way and we could [colliding with another symbol](https://github.com/libgit2/libgit2/issues/3825).

`git_varint_encode` and `git_varint_decode` would be my recommendations.